src/shared/switch-root.c

```diff
@@ -112,6 +118,9 @@ int switch_root(const char *new_root,
         FOREACH_ARRAY(transfer, transfer_table, ELEMENTSOF(transfer_table)) {
                 _cleanup_free_ char *chased = NULL;
 
+                if (transfer->skip_if_run_is_rec && !FLAGS_SET(flags, SWITCH_ROOT_NON_RECURSIVE_RUN))
```

Comment:
The third flag is not necessary,
```
if (!isempty(path_startswith(transfer->path, "/run/")) && FLAGS_SET(flags, SWITCH_ROOT_RECURSIVE_RUN))
```

Reply:
(`isempty` is unneeded too. `path_startswith` returns `NULL` if not matched.)

Reply:
better to encode it explicitly than to rely on the precise path, as it will be harder to get it wrong

Reply:
> (`isempty` is unneeded too. `path_startswith` returns `NULL` if not matched.)

`isempty()` is for matching all sub-mounts, but not "/run/" itself.

Reply:
> better to encode it explicitly than to rely on the precise path, as it will be harder to get it wrong

When we add a new feature that uses another /run/ submount, then we also need to set the flag.
Table can be easily outdated, and redundant table fields should not exist in general.
I do not think the suggested code is not complicated, Adding a short comment is enough.

Reply:
No, I prefer to have the explicit flag, it's clearer

Reply:
I'd vote for the flag too. It's more explicit and clearer.